src/query.rs

```diff
@@ -226,6 +218,20 @@ impl QueryExecutor {
         let column_types = Arc::clone(&self.column_types);
 
         tokio::spawn(async move {
+            // First process rows sent along with the initial results response
+            if let Some(row_set) = row_set.take() {
+                for row in row_set.into_iter() {
+                    let _ = tx
+                        .send(Ok(SnowflakeRow {
+                            row,
+                            column_indices: Arc::clone(&column_indices),
+                            column_types: Arc::clone(&column_types),
+                        }))
+                        .await;
+                }
+            }
+            let _ = row_set;
```

Comment:
**Nitpick (non-blocking)**: you could also just do `drop(row_set)`, but I'm completely fine with `let _` (I didn't not ice it before even)